src/frontend/hooks/useDraftObservation.tsx

```diff
@@ -33,38 +32,40 @@ const PREVIEW_QUALITY = 30;
 type Signal = { didCancel: boolean };
 
 export type CapturedPictureMM = {
-  uri: string,
-  rotate?: number,
+  uri: string;
+  rotate?: number;
 };
 
 export type UseDraftObservation = [
-  {|
-    value: $ElementType<DraftObservationContextState, "value">,
-    photos: $ElementType<DraftObservationContextState, "photos">,
-    observationId: $ElementType<DraftObservationContextState, "observationId">,
-    savingStatus: Status,
-    preset?: PresetWithFields,
-  |},
-  {|
+  {
+    value: DraftObservationContextState["value"];
+    photos: DraftObservationContextState["photos"];
+    observationId: DraftObservationContextState["observationId"];
+    savingStatus: Status;
+    preset?: PresetWithFields;
+  },
+  {
     /**
      * Adds a photo to the draft observation. The first argument is a promise
      * which returns a uri to a local image file (in temp cache) and the image
      * width and height.
      */
-    addPhoto: (capture: Promise<CapturedPictureMM>) => Promise<void>,
+    addPhoto: (capture: Promise<CapturedPictureMM>) => Promise<void>;
     // Save a draft
-    saveDraft: () => Promise<void>,
+    saveDraft: () => Promise<void>;
     // Performs a shallow merge of the observation value, like setState
-    updateDraft: (value: ObservationValue) => void,
+    updateDraft: (value: ObservationValue) => void;
     // Clear the current draft
-    clearDraft: () => void,
+    clearDraft: () => void;
     // Create a new draft observation
     newDraft: (
       id?: string,
       value?: ObservationValue | null,
       capture?: Promise<CapturedPictureMM>
-    ) => void,
-  |}
+    ) => void;
+
+    deletePhoto: (index: number) => void;
```

Comment:
is it too difficult/awkward to use some unique identifier on the photo for the argument here? probably not fully necessary but always a little hesitant about using indices for this kind of thing (maybe unreasonably so 😅 )

Reply:
Not unreasonable. I am just pulling the uri, using the index. Kind of an extra step, but it would probably be more robust in the long term!